### DISCUSSION TITLE
Your branch and 'origin/master' have diverged?

### DISCUSSION THREAD

**Discussion by jmmb** [community/community#22704] [New to GitHub]
<p>Hello - I have a rudimentary understanding of Git, and recently pushed a local repository to GH Pages (to create a simple static HTML webpage).</p>
<p>After the initial commit, I made one text change to the “index.html” file and pushed that to GH Pages via the standard ‘git push’ command. This worked perfectly, and I thought everything was clear with using GitHub for this website …</p>
<p>However, yesterday I made another text change to “index.html” on my local machine, and tried the same method to push it up. To my complete surprise, I got a message saying that one branch was behind the other (something to that effect), and that I should do ‘git pull’ to sort it out.</p>
<p>Unfortunately, that didn’t work either (it happened fast and I’m a little fuzzy about the specifics), but I ended up aborting the process and nothing was committed.</p>
<p>So today I figured I should check the status, so I did ‘git status’ and here was the result:</p>
<blockquote>
<p>On branch master<br>
Your branch and ‘origin/master’ have diverged,<br>
and have 2 and 1 different commits each, respectively.<br>
(use “git pull” to merge the remote branch into yours)<br>
nothing to commit, working tree clean</p>
</blockquote>
<p>So then I did ‘git pull’, with this result:</p>
<blockquote>
<p>Swap file “~/Sites/www.myname.com/.git/.MERGE_MSG.swp” already exists!<br>
[O]pen Read-Only, (E)dit anyway, ®ecover, (D)elete it, (Q)uit, (A)bort:<br>
E325: ATTENTION<br>
Found a swap file by the name “~/Sites/www.myname.com/.git/.MERGE_MSG.swp”<br>
owned by: JimB   dated: Fri Aug 14 18:56:48 2020<br>
file name: ~JimB/Sites/www.myname.com/.git/MERGE_MSG<br>
modified: YES<br>
user name: JimB   host name: Jim-Bs-MacBook-Pro.local<br>
process ID: 1918<br>
While opening file “/Users/JimB/Sites/www.myname.com/.git/MERGE_MSG”<br>
dated: Sat Aug 15 22:00:32 2020<br>
NEWER than swap file!</p>
<p>(1) Another program may be editing the same file.  If this is the case,<br>
be careful not to end up with two different instances of the same<br>
file when making changes.  Quit, or continue with caution.<br>
(2) An edit session for this file crashed.<br>
If this is the case, use “:recover” or “vim -r /Users/JimB/Sites/www.myname.com/.git/MERGE_MSG”<br>
to recover the changes (see “:help recovery”).<br>
If you did this already, delete the swap file “/Users/JimB/Sites/www.myname.com/.git/.MERGE_MSG.swp”<br>
to avoid this message.</p>
</blockquote>
<p>Note: “<a href="http://myname.com" rel="nofollow noopener">myname.com</a>” is not the actual directory name …</p>
<p>Anyway, I’m pretty confident that (1) is not the case here. I’m the only person who would know about or have access to this repository.</p>
<p>So I suppose that leaves (2) as the most likely explanation. Unfortunately, I have no idea what this means. I’ve used GitHub for a long time just doing simple commits like this one, and have never seen anything like this before.</p>
<p>So then given the choices, I chose “A” to abort, which yielded this:</p>
<blockquote>
<p>error: There was a problem with the editor ‘vi’.<br>
Not committing merge; use ‘git commit’ to complete the merge.<br>
JimB@Jim-Bs-MacBook-Pro <a href="http://www.myname.com" rel="nofollow noopener">www.myname.com</a> %</p>
</blockquote>
<p>I’m pretty confused about what’s going on here, and definitely don’t know what to do next. And since I don’t want to mess anything up, I’m asking this question here.</p>
<p>Thanks for any assistance!</p>

**Comment by jmmb** [comment, marked as answer]
<p>Just a quick update that I don’t need help anymore (though it’s unclear if any help was coming) …</p>
<p>I decided to follow the topmost answer at <a href="https://stackoverflow.com/questions/19864934/git-your-branch-and-origin-master-have-diverged-how-to-throw-away-local-com" rel="nofollow noopener">https://stackoverflow.com/questions/19864934/git-your-branch-and-origin-master-have-diverged-how-to-throw-away-local-com</a></p>
<p>And that fixed the problem in terms of resetting my local directory to match the master branch on GH Pages.</p>
<p>And then I realized there still needed to be a small text change to the “index.html” file, so I just changed that file locally and did ‘git add .’ plus ‘git commit -m “message”’ and then ‘git push’ …</p>
<p>It all worked perfectly, so hopefully everything will work now as expected going forward.</p>

**Reply by chianugo** [reply]
This helped me with a similar issue where I deleted the CNAME through git hub pages and tried to push an update from my machine. Thank you.
